src/Cli/dotnet/PerformanceLogEventSource.cs

```diff
@@ -0,0 +1,461 @@
+﻿using System;
+using System.Diagnostics;
+using System.Diagnostics.Tracing;
+using System.IO;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using Microsoft.DotNet.Cli.Utils;
+using RuntimeEnvironment = Microsoft.DotNet.Cli.Utils.RuntimeEnvironment;
+
+namespace Microsoft.DotNet.Cli
+{
+    [EventSource(Name = "Microsoft-Dotnet-CLI-Performance", Guid = "cbd57d06-3b9f-5374-ed53-cfbcc23cf44f")]
+    internal sealed class PerformanceLogEventSource : EventSource
+    {
+        internal static PerformanceLogEventSource Log = new PerformanceLogEventSource();
+
+        private PerformanceLogEventSource()
+        {
+        }
+
+        [NonEvent]
+        internal void LogStartUpInformation(PerformanceLogStartupInformation startupInfo)
+        {
+            if(!IsEnabled())
+            {
+                return;
+            }
+
+            DotnetVersionFile versionFile = DotnetFiles.VersionFileObject;
+            string commitSha = versionFile.CommitSha ?? "N/A";
+
+            LogMachineConfiguration();
+            OSInfo(RuntimeEnvironment.OperatingSystem, RuntimeEnvironment.OperatingSystemVersion, RuntimeEnvironment.OperatingSystemPlatform.ToString());
+            SDKInfo(Product.Version, commitSha, GetDisplayRid(versionFile), AppContext.BaseDirectory);
+            EnvironmentInfo(Environment.CommandLine);
+            LogMemoryConfiguration();
+            LogDrives();
+
+            // It's possible that IsEnabled returns true if an out-of-process collector such as ETW is enabled.
+            // If the perf log hasn't been enabled, then startupInfo will be null, so protect against nullref here.
+            if (startupInfo != null)
+            {
+                if (startupInfo.TimedAssembly != null)
+                {
+                    AssemblyLoad(startupInfo.TimedAssembly.GetName().Name, startupInfo.AssemblyLoadTime.TotalMilliseconds);
+                }
+
+                Process currentProcess = Process.GetCurrentProcess();
+                TimeSpan latency = startupInfo.MainTimeStamp - currentProcess.StartTime;
+                HostLatency(latency.TotalMilliseconds);
+            }
+        }
+
+        [NonEvent]
+        private static string GetDisplayRid(DotnetVersionFile versionFile)
+        {
+            FrameworkDependencyFile fxDepsFile = new FrameworkDependencyFile();
+
+            string currentRid = System.Runtime.InteropServices.RuntimeInformation.RuntimeIdentifier;
+
+            // if the current RID isn't supported by the shared framework, display the RID the CLI was
+            // built with instead, so the user knows which RID they should put in their "runtimes" section.
+            return fxDepsFile.IsRuntimeSupported(currentRid) ?
```

Comment:
nit: maybe we should just log currentRid and build rid separately, so we don't need to do guess work while analyses

Reply:
Makes sense.  Done.